airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2052,6 +2053,8 @@ class BigQueryInsertJobOperator(BaseOperator):
         Service Account Token Creator IAM role to the directly preceding identity, with first
         account from the list granting this role to the originating account (templated).
     :param cancel_on_kill: Flag which indicates whether cancel the hook's job or not, when on_kill is called
+    :param retry: Designation of what errors, if any, should be retried.
```

Comment:
This may be very confusing with `retries` from `BaseOprerator`.

```
BigQueryInsertJobOperator(
...,
retry=3,
retries=10
)
```

We need better name for it - I think. Something that the user will know that retries refer to BigQuery "internal" capability.

I would say that the same goes for the `timeout`.

Reply:
👍

Reply:
@eladkal updated

Reply:
To clarify my comment is on the Operator. We don't have this problem in the hook.

Reply:
@eladkal I know but I wanted to keep the same names of parameters in both places to avoid problems with understanding.

Reply:
I appreciate that but note also that other functions in the hook has this parameter already defined as retry and also with default of `DEFAULT_RETRY`

https://github.com/apache/airflow/blob/71c980a8ffb3563bf16d8a23a58de54c9e8cf556/airflow/providers/google/cloud/hooks/bigquery.py#L320

https://github.com/apache/airflow/blob/71c980a8ffb3563bf16d8a23a58de54c9e8cf556/airflow/providers/google/cloud/hooks/bigquery.py#L478

I would suggest to check how/if other operators utilize the retry in the other functions it should help to make a decision what is the best approach here. I don't feel strongly about any specific approach just want to make sure that we are consistent (or if we want to change conventions that is also ok - just require more work)